kotlin/codegen/build.gradle.kts

```diff
@@ -103,8 +112,21 @@ afterEvaluate {
       if (name == "pluginMaven") {
         // This is to properly wire the shadow jar's gradle metadata and pom information
         setArtifacts(artifacts.matching { it.classifier != "" })
-        // Ugly but artifact() doesn't support TaskProviders
-        artifact(shadowJar.get())
+        artifact(shadowJar)
+      }
+
+      pom.withXml {
+        // Fix runtime scopes being put here by the shadow plugin
+        // https://github.com/johnrengelman/shadow/issues/321
+        val xmlSb = asString()
+        val runtime = ">runtime<"
+        val replacement = ">compile<"
+        xmlSb.toString()
+          .findAllIndicesOf(">runtime<")
+          .reversed()
+          .forEach { index ->
+            xmlSb.replace(index, index + runtime.length, replacement)
+          }
```

Comment:
This may actually be resolvable by just making them `api` dependencies. @gabrielittner pointed out privately that this may be a kapt bug. I'll try with just making them `api`

> my guess is that it's just an issue for annotation processors, maybe it's even kapt specific
>> if the library is published with Gradle, the produced POM file only puts `api` dependencies into the compile scope and the remaining `implementation` dependencies into the runtime scope.
>
> https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

Reply:
That worked 👍

Reply:
Meanwhile filed here: https://youtrack.jetbrains.com/issue/KT-41702